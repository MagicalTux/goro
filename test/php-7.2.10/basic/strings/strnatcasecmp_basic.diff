--
     int(1)
     int(-1)
     int(1)
007- int(0)
008- int(1)
009- int(1)
010- int(-1)
011- int(-1)
012- int(0)
013- int(-1)
014- int(0)
015- ===DONE===
007+ panic: runtime error: index out of range [3] with length 3
008+ 
009+ goroutine 1 [running]:
010+ github.com/MagicalTux/goro/ext/standard.natCmp({0xc000207cd9, 0x3, 0x3}, {0xc000207cdc, 0x3, 0x3}, 0x0)
011+ 	/home/ronald/proj/goro/ext/standard/natcmp.go:34 +0x52d
012+ github.com/MagicalTux/goro/ext/standard.fncStrNatCaseCmp({0x8bd150, 0xc00021ca00}, {0xc000412d30, 0x2, 0x2})
013+ 	/home/ronald/proj/goro/ext/standard/strings.go:1573 +0x335
014+ github.com/MagicalTux/goro/core/phpctx.(*ExtFunction).Call(0xc0001ad0e0, {0x8bd150, 0xc00021ca00}, {0xc000412d30, 0x2, 0x2})
015+ 	/home/ronald/proj/goro/core/phpctx/ext.go:33 +0x9d
016+ github.com/MagicalTux/goro/core/phpctx.(*Global).CallZVal(0xc00025c000, {0x8bd150, 0xc00021c980}, {0x8badf0, 0xc0001ad0e0}, {0xc000412d30, 0x2, 0x2}, {0xc000412d00, 0x1, ...})
017+ 	/home/ronald/proj/goro/core/phpctx/global-call.go:84 +0x8a6
018+ github.com/MagicalTux/goro/core/phpctx.(*Global).Call(0xc00025c000, {0x8bd150, 0xc00021c980}, {0x8badf0, 0xc0001ad0e0}, {0xc000203620, 0x2, 0x2}, {0xc000412d00, 0x1, ...})
019+ 	/home/ronald/proj/goro/core/phpctx/global-call.go:18 +0x32c
020+ github.com/MagicalTux/goro/core/compiler.(*runnableFunctionCallRef).Run(0xc00040e9f0, {0x8bd150, 0xc00021c980})
021+ 	/home/ronald/proj/goro/core/compiler/compile-func.go:160 +0x1590
022+ github.com/MagicalTux/goro/core/phpctx.(*Global).Call(0xc00025c000, {0x8bd150, 0xc00021c980}, {0x8badf0, 0xc0001ad680}, {0xc000412290, 0x1, 0x1}, {0xc000412ce0, 0x1, ...})
023+ 	/home/ronald/proj/goro/core/phpctx/global-call.go:12 +0x14e
024+ github.com/MagicalTux/goro/core/compiler.(*runnableFunctionCallRef).Run(0xc00040ea20, {0x8bd150, 0xc00021c980})
025+ 	/home/ronald/proj/goro/core/compiler/compile-func.go:160 +0x1590
026+ github.com/MagicalTux/goro/core/phpv.Runnables.Run({0xc0004122a0, 0x1, 0x1}, {0x8bd150, 0xc00021c980})
027+ 	/home/ronald/proj/goro/core/phpv/run.go:9 +0x103
028+ github.com/MagicalTux/goro/core/compiler.(*ZClosure).Call(0xc00021c080, {0x8bd150, 0xc00021c980}, {0xc000412c70, 0x2, 0x2})
029+ 	/home/ronald/proj/goro/core/compiler/closure.go:197 +0x5fc
030+ github.com/MagicalTux/goro/core/phpctx.(*Global).CallZVal(0xc00025c000, {0x8bd028, 0xc00025c000}, {0x8baf30, 0xc00021c080}, {0xc000412c70, 0x2, 0x2}, {0xc000412c40, 0x1, ...})
031+ 	/home/ronald/proj/goro/core/phpctx/global-call.go:84 +0x8a6
032+ github.com/MagicalTux/goro/core/phpctx.(*Global).Call(0xc00025c000, {0x8bd028, 0xc00025c000}, {0x8baf30, 0xc00021c080}, {0xc0003ac7c0, 0x2, 0x2}, {0xc000412c40, 0x1, ...})
033+ 	/home/ronald/proj/goro/core/phpctx/global-call.go:18 +0x32c
034+ github.com/MagicalTux/goro/core/compiler.(*runnableFunctionCallRef).Run(0xc0003be120, {0x8bd028, 0xc00025c000})
035+ 	/home/ronald/proj/goro/core/compiler/compile-func.go:160 +0x1590
036+ github.com/MagicalTux/goro/core/phpv.Runnables.Run({0xc0001b4100, 0x10, 0x10}, {0x8bd028, 0xc00025c000})
037+ 	/home/ronald/proj/goro/core/phpv/run.go:9 +0x103
038+ github.com/MagicalTux/goro/core/phpctx.(*Global).Require(0xc00025c000, {0x8bd028, 0xc00025c000}, {0x7fff434ca718, 0x4c})
039+ 	/home/ronald/proj/goro/core/phpctx/global-inc.go:116 +0x4da
040+ github.com/MagicalTux/goro/core/phpctx.(*Global).RunFile(0xc00025c000, {0x7fff434ca718, 0x4c})
041+ 	/home/ronald/proj/goro/core/phpctx/global.go:246 +0x7f
042+ main.main()
043+ 	/home/ronald/proj/goro/sapi/php-cli/main.go:60 +0x6a9
