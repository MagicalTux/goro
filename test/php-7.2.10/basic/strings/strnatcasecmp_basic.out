*** Testing strnatcasecmp() : basic functionality ***
int(0)
int(-1)
int(1)
int(-1)
int(1)
panic: runtime error: index out of range [3] with length 3

goroutine 1 [running]:
github.com/MagicalTux/goro/ext/standard.natCmp({0xc000207cd9, 0x3, 0x3}, {0xc000207cdc, 0x3, 0x3}, 0x0)
	/home/ronald/proj/goro/ext/standard/natcmp.go:34 +0x52d
github.com/MagicalTux/goro/ext/standard.fncStrNatCaseCmp({0x8bd150, 0xc00021ca00}, {0xc000412d30, 0x2, 0x2})
	/home/ronald/proj/goro/ext/standard/strings.go:1573 +0x335
github.com/MagicalTux/goro/core/phpctx.(*ExtFunction).Call(0xc0001ad0e0, {0x8bd150, 0xc00021ca00}, {0xc000412d30, 0x2, 0x2})
	/home/ronald/proj/goro/core/phpctx/ext.go:33 +0x9d
github.com/MagicalTux/goro/core/phpctx.(*Global).CallZVal(0xc00025c000, {0x8bd150, 0xc00021c980}, {0x8badf0, 0xc0001ad0e0}, {0xc000412d30, 0x2, 0x2}, {0xc000412d00, 0x1, ...})
	/home/ronald/proj/goro/core/phpctx/global-call.go:84 +0x8a6
github.com/MagicalTux/goro/core/phpctx.(*Global).Call(0xc00025c000, {0x8bd150, 0xc00021c980}, {0x8badf0, 0xc0001ad0e0}, {0xc000203620, 0x2, 0x2}, {0xc000412d00, 0x1, ...})
	/home/ronald/proj/goro/core/phpctx/global-call.go:18 +0x32c
github.com/MagicalTux/goro/core/compiler.(*runnableFunctionCallRef).Run(0xc00040e9f0, {0x8bd150, 0xc00021c980})
	/home/ronald/proj/goro/core/compiler/compile-func.go:160 +0x1590
github.com/MagicalTux/goro/core/phpctx.(*Global).Call(0xc00025c000, {0x8bd150, 0xc00021c980}, {0x8badf0, 0xc0001ad680}, {0xc000412290, 0x1, 0x1}, {0xc000412ce0, 0x1, ...})
	/home/ronald/proj/goro/core/phpctx/global-call.go:12 +0x14e
github.com/MagicalTux/goro/core/compiler.(*runnableFunctionCallRef).Run(0xc00040ea20, {0x8bd150, 0xc00021c980})
	/home/ronald/proj/goro/core/compiler/compile-func.go:160 +0x1590
github.com/MagicalTux/goro/core/phpv.Runnables.Run({0xc0004122a0, 0x1, 0x1}, {0x8bd150, 0xc00021c980})
	/home/ronald/proj/goro/core/phpv/run.go:9 +0x103
github.com/MagicalTux/goro/core/compiler.(*ZClosure).Call(0xc00021c080, {0x8bd150, 0xc00021c980}, {0xc000412c70, 0x2, 0x2})
	/home/ronald/proj/goro/core/compiler/closure.go:197 +0x5fc
github.com/MagicalTux/goro/core/phpctx.(*Global).CallZVal(0xc00025c000, {0x8bd028, 0xc00025c000}, {0x8baf30, 0xc00021c080}, {0xc000412c70, 0x2, 0x2}, {0xc000412c40, 0x1, ...})
	/home/ronald/proj/goro/core/phpctx/global-call.go:84 +0x8a6
github.com/MagicalTux/goro/core/phpctx.(*Global).Call(0xc00025c000, {0x8bd028, 0xc00025c000}, {0x8baf30, 0xc00021c080}, {0xc0003ac7c0, 0x2, 0x2}, {0xc000412c40, 0x1, ...})
	/home/ronald/proj/goro/core/phpctx/global-call.go:18 +0x32c
github.com/MagicalTux/goro/core/compiler.(*runnableFunctionCallRef).Run(0xc0003be120, {0x8bd028, 0xc00025c000})
	/home/ronald/proj/goro/core/compiler/compile-func.go:160 +0x1590
github.com/MagicalTux/goro/core/phpv.Runnables.Run({0xc0001b4100, 0x10, 0x10}, {0x8bd028, 0xc00025c000})
	/home/ronald/proj/goro/core/phpv/run.go:9 +0x103
github.com/MagicalTux/goro/core/phpctx.(*Global).Require(0xc00025c000, {0x8bd028, 0xc00025c000}, {0x7fff434ca718, 0x4c})
	/home/ronald/proj/goro/core/phpctx/global-inc.go:116 +0x4da
github.com/MagicalTux/goro/core/phpctx.(*Global).RunFile(0xc00025c000, {0x7fff434ca718, 0x4c})
	/home/ronald/proj/goro/core/phpctx/global.go:246 +0x7f
main.main()
	/home/ronald/proj/goro/sapi/php-cli/main.go:60 +0x6a9